app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx

```diff
@@ -2,25 +2,37 @@ import React from "react";
 
 import { Formik, Form as FormikForm } from "formik";
 import { Button, Pane } from "neetoui";
-import { Input, Textarea } from "neetoui/formik";
+import { Input, Textarea, Select } from "neetoui/formik";
 
-import notesApi from "apis/notes";
+import {
+  NOTES_FORM_VALIDATION_SCHEMA,
+  AVAILABLE_NOTE_TAGS,
+  DUMMY_CONTACTS,
+} from "../constants";
 
-import { NOTES_FORM_VALIDATION_SCHEMA } from "../constants";
+const Form = ({ onClose, note, isEdit, setNotes }) => {
+  const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
 
-const Form = ({ onClose, refetch, note, isEdit }) => {
-  const handleSubmit = async values => {
-    try {
+  const handleSubmit = async value => {
+    await delay(500);
+
+    const noteData = { ...value };
+    noteData.status = "Created";
+
+    setNotes(prevNotes => {
       if (isEdit) {
-        await notesApi.update(note.id, values);
-      } else {
-        await notesApi.create(values);
+        const index = prevNotes.findIndex(note => note.id === noteData.id);
+        prevNotes[index] = noteData;
+
+        return prevNotes;
       }
-      refetch();
-      onClose();
-    } catch (err) {
-      logger.error(err);
-    }
+      noteData.added_at = new Date().toLocaleDateString();
+      noteData.id = prevNotes[0].id + 1;
+
+      return [noteData, ...prevNotes];
```

Comment:
Suppose two or more processes simultaneously try to add a new note to the list of `prevNotes`. If they both read the value of `prevNotes[0].id` before any of them have a chance to update the list, they will get the same `prevNotes[0].id` value.

Let's use [uuid](https://www.npmjs.com/package/uuid) to create unique identifiers.

Reply:
Yes, I have started using uuid while working with `contacts` page, will update this as well.
Thanks.